mne_bids/path.py

```diff
@@ -405,6 +405,25 @@ def update(self, check=None, **entities):
         self._check()
         return self
 
+    def match(self):
+        """Get a list of all matching basenames.
+
+        Performs a search based on the entities that are set in the
+        ``BIDSPath`` object. Requires a ``root`` attribute to be set.
+
+        Returns
+        -------
+        paths : List[BIDSPath]
+            All paths with matching basenames as a BIDSPath.
+        """
+        if self.root is None:
```

Comment:
Why? What if I want to use the CWD? must I set `root='.'` then? If that's the case, we should either change the default of `BIDSPath.root` to `'.'`, or set it to `'.'` if `root is None`

Reply:
Okay that's pretty sensible, so root is either `'.'` (either set manually, or `None`), or some set path.

Reply:
Yeah… now I'm not so sure anymore. Help! I keep changing my mind!!
Maybe let's leave it the way it is for now, and rethink it later? There's probably also other places where we would need to adjust this.